### PR TITLE
feat(replace NATS): add watcher operations over grpc replacing NATS

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**
Enable Watcher ops over GRPC, replacing NATS.

Signed-off-by: Abhinandan Purkait <abhinandan.purkait@mayadata.io>

### CODE REVIEW

**Comment by Abhinandan-Purkait** [comment]
bors try

File: control-plane/grpc/src/operations/watcher/server.rs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    /// Service which executes the operations.
+    service: Arc<dyn WatcherOperations>,
+}
+
+impl WatcherServer {
+    /// returns a new WatcherServer with the service implementing Watcher operations
+    pub fn new(service: Arc<dyn WatcherOperations>) -> Self {
+        Self { service }
+    }
+    /// coverts the replicaserver to its corresponding grpc server type
```
Comment (paulyoong) [inline control-plane/grpc/src/operations/watcher/server.rs:26]: Nit: Typo "converts". Also `replicaserver` should be `WatcherServer`.

**Review by paulyoong — APPROVED** [review]

File: control-plane/grpc/src/operations/watcher/server.rs (hunk 1 above)
Comment (Abhinandan-Purkait) [reply control-plane/grpc/src/operations/watcher/server.rs]: Done

File: control-plane/grpc/proto/v1/watcher/watcher.proto [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  // id of the resource to watch on
+  WatchResourceId id = 1;
+  // callback used to notify the watcher of a change
+  WatchCallback callback = 2;
+  // type of watch
+  WatchType watch_type = 3;
+}
+
+// Multiple Watch
+message Watches { repeated Watch watches = 1; }
```
[the remark this answers is not in the document]
Comment (tiagolobocastro) [reply control-plane/grpc/proto/v1/watcher/watcher.proto]: This was likely my fault for using both watchers and watches...

**Review by tiagolobocastro — APPROVED** [review]

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Pull request merged**